### PR TITLE
Follow-up on ExprValueFactory deprecation and silence other compile warnings

### PR DESCRIPTION
**Pull request by vgapeyev**
Some tidying up following the deprecation, in PR944, of ExprValueFactory, in favor of new value constructors in ExprValue. (The primary purpose of that work was simplifications enabled by the removal, in PR928, of ExprValue dependency on IonSystem.)

The major thing here was silencing related compilation warnings, so that they do not litter the build log.

The commits are fairly self-contained and focused, if that aids reviewing. 

## Other Information
- Updated Unreleased Section in CHANGELOG: **[NO]**
  - internal clean-up for the most part;
  - things that can be detected externally are too fiddly to describe
- Any backward-incompatible changes? **[NO]**
- Any new external dependencies? **[NO]**

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt [hunk 1]
```diff
@@ -41,7 +40,7 @@ import org.partiql.lang.util.interruptibleFold
 data class StepContext(
     @Deprecated("[ExprValueFactory] is deprecated")
     /** The instance of [ExprValueFactory] that is used by the pipeline. */
-    val valueFactory: ExprValueFactory,
+    val valueFactory: org.partiql.lang.eval.ExprValueFactory,
```
Comment (RCHowell) [inline lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt]: Why did this need to be full qualified?

**Review by RCHowell — CHANGES_REQUESTED** [review]

**Comment by vgapeyev** [comment]
The class ExprValueFactory is deprecated, but it does not seem to possible to mark its file-level import with ***@***.***(“DEPRECATION”)`.

From: "R. C. Howell" ***@***.***>
Reply-To: partiql/partiql-lang-kotlin ***@***.***>
Date: Monday, February 27, 2023 at 11:29 AM
To: partiql/partiql-lang-kotlin ***@***.***>
Cc: "Gapeyev, Vladimir" ***@***.***>, Author ***@***.***>
Subject: Re: [partiql/partiql-lang-kotlin] Follow-up on ExprValueFactory deprecation (PR #985)


@RCHowell requested changes on this pull request.

________________________________

In lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt<https://github.com/partiql/partiql-lang-kotlin/pull/985#discussion_r1119199589>:

> @@ -41,7 +40,7 @@ import org.partiql.lang.util.interruptibleFold

 data class StepContext(

     @Deprecated("[ExprValueFactory] is deprecated")

     /** The instance of [ExprValueFactory] that is used by the pipeline. */

-    val valueFactory: ExprValueFactory,

+    val valueFactory: org.partiql.lang.eval.ExprValueFactory,

Why did this need to be full qualified?

File: lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt [hunk 2]
```diff
@@ -21,7 +21,6 @@ import org.partiql.lang.eval.Bindings
 import org.partiql.lang.eval.CompileOptions
 import org.partiql.lang.eval.EvaluatingCompiler
 import org.partiql.lang.eval.ExprFunction
-import org.partiql.lang.eval.ExprValueFactory
```
Comment (RCHowell) [inline lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt:24]: I mean to say, why was this import removed and ExprValueFactory fully qualified? `Suppress("DEPRECATION")` seems to work fine without needing to fully qualify ExprValueFactory

File: lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt [hunk 3]
```diff
@@ -116,27 +125,30 @@ interface CompilerPipeline {
...
 
         /** Fluent style builder.  If calling from Kotlin instead use the [build] method. */
         @JvmStatic
         fun builder(): Builder = Builder()
 
-        @Deprecated("[ExprValueFactory] is deprecated. Please use `builder(ion: IonSystem): Builder = builder(ion)`.")
         /** Fluent style builder.  If calling from Kotlin instead use the [build] method. */
         @JvmStatic
-        fun builder(valueFactory: ExprValueFactory): Builder = Builder(valueFactory)
+        @Deprecated("[ExprValueFactory] is deprecated. Please use `builder(): Builder = builder(ion)`.")
```
Comment (RCHowell) [inline lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt:139]: '' `ReplaceWith`

File: lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt [hunk 4]
```diff
@@ -116,27 +125,30 @@ interface CompilerPipeline {
...
 
         /** Returns an implementation of [CompilerPipeline] with all properties set to their defaults. */
         @JvmStatic
         fun standard(): CompilerPipeline = builder().build()
 
-        @Deprecated("[ExprValueFactory] is deprecated. Please use `standard(ion: IonSystem): CompilerPipeline`.")
         /** Returns an implementation of [CompilerPipeline] with all properties set to their defaults. */
         @JvmStatic
-        fun standard(valueFactory: ExprValueFactory): CompilerPipeline = builder(valueFactory).build()
+        @Deprecated("[ExprValueFactory] is deprecated. Please use `standard(): CompilerPipeline`.")
```
Comment (RCHowell) [inline lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt:149]: '' `ReplaceWith`

**Review by RCHowell — CHANGES_REQUESTED** [review]

File: lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt (hunk 2 above)
Comment (vgapeyev) [reply lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt:24]: Sorry for that email-response.  

So, when this file-level import is present, it gives a warning that it imports a deprecated class.  I do not know how to suppress this warning:  `@Suppress("DEPRECATION")` is not allowed here.  Hence, fully-qualified in all places where the deprecated class needs to be mentioned.

**Comment by vgapeyev** [comment]
Robert's comments addressed.

**Review by RCHowell — APPROVED** [review]
